### PR TITLE
Add current visitor location doc

### PR DESCRIPTION
**Pull request by ostap0207**
CICT-14

https://github.com/salemove/web_services/pull/733

### CODE REVIEW

**Comment by ostap0207** [comment]
@indrekj @urmastalimaa @deiwin

File: apiary.apib [hunk 1]
```diff
@@ -1246,6 +1246,14 @@ Fetches the information of current visitor on the site.
 
 + Response 200 (application/json)
 
+    + Attributes
+      + location (required, object) - IP based current visitor location
```
Comment (indrekj) [inline apiary.apib]: should it be under `Attributes`. What exactly is `required` here?

Comment (ostap0207) [inline apiary.apib]: I guess it their way of saying that we gonna 100% return this parameter. Not sure though. Here is the doc https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#def-attributes-section

Comment (indrekj) [inline apiary.apib]: I think just remove the `required, ` part. Then it looks okay

Comment (ostap0207) [inline apiary.apib]: I've tried, but in the UI it says `optional` in this case. Which is also confusing. IMHO would be `required` better.

File: apiary.apib [hunk 2]
```diff
@@ -1246,6 +1246,14 @@ Fetches the information of current visitor on the site.
 
 + Response 200 (application/json)
 
+    + Attributes
```
Comment (deiwin) [inline apiary.apib]: I don't see `Response -> Attributes` structure used elsewhere in the doc. Does this compile okay?

Comment (deiwin) [inline apiary.apib]: Also, should probably document the rest of the attributes as well if you're already documenting some.

Comment (ostap0207) [inline apiary.apib]: I don't thing we've used it. But it compiles OK https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#example-16.

**Comment by deiwin** [comment]
Do you know what the current use cases are for the endpoint? We might want to keep the endpoint more minimal if it is hit often

File: apiary.apib (hunk 2 above)
Comment (ostap0207) [inline apiary.apib]: It's says it's ok not document everything. I just added it for location to specify the standard. Other parameters are self-explanatory.

**Comment by ostap0207** [comment]
Don't know, @urmastalimaa said that it might not be used much.

File: apiary.apib (hunk 2 above)
Comment (deiwin) [inline apiary.apib]: Fair enough

**Comment by urmastalimaa** [comment]
No I don't believe I said that.
I said that it's conceivable that the code for endpoint users might break when adding attributes to the response, so we might want to think about that.

**Comment by ostap0207** [comment]
Is there a way to find more about traffic through our api?

**Comment by ostap0207** [comment]
Alternatively can add endpoint `/visitor/location` instead. What do you think?

**Comment by deiwin** [comment]
You mean `/visitors/:id/location`, right? I think that would be a better approach, really, because I don't see the information being all that useful in the general endpoint.

**Comment by urmastalimaa** [comment]
Agree

**Comment by ostap0207** [comment]
Ok. @deiwin The endpoint is `/visitor/location`. It gets the information for the current visitor from visitor side.

**Comment by deiwin** [comment]
Ah, okay. That's alright then

**Comment by ostap0207** [comment]
This has been updated

File: apiary.apib [hunk 3]
```diff
@@ -1306,6 +1306,36 @@ The ```note_update_method``` parameter takes either value ```replace``` which re
           }
         }
 
+## Current visitor location [/visitor/location]
+
+### Get current visitor location [GET]
+
+Fetches the location of a current visitor on the site.
```
Comment (deiwin) [inline apiary.apib]: `a` -> `the`

Comment (deiwin) [inline apiary.apib]: `the location` -> `the geographic location` I think

Comment (deiwin) [inline apiary.apib]: and remove ` on the site`. I think it can only add confusion

Comment (ostap0207) [inline apiary.apib]: I would still maybe leave at list `of the current site visitor` to make more distinction with other endpoint `/visitor/{visitor_id}`

Comment (deiwin) [inline apiary.apib]: Maybe reword it to say: `Fetches the current visitor's own geographical location.` This way there shouldn't be any confusion with the endpoints with the ID argument.

**Comment by ostap0207** [comment]
Updated

File: apiary.apib [hunk 4]
```diff
@@ -1306,6 +1306,36 @@ The ```note_update_method``` parameter takes either value ```replace``` which re
           }
         }
 
+## Current visitor location [/visitor/location]
+
+### Get current visitor location [GET]
+
+Fetches the current visitor's own geographical location.
```
Comment (willmore) [inline apiary.apib]: `own` is redundant.

File: apiary.apib [hunk 5]
```diff
@@ -1306,6 +1306,36 @@ The ```note_update_method``` parameter takes either value ```replace``` which re
           }
         }
 
+## Current visitor location [/visitor/location]
```
Comment (willmore) [inline apiary.apib]: Use of the term 'Current' is ambiguous. Current denotes relationship in time. The system can have many "current" users.
In the context of the API interaction, the only relationship that determines the response, is that of the requestor. So, what I think we are trying to say is 'Requesting visitor location'.
Instead of "current", we should use "requesting".

Comment (ostap0207) [inline apiary.apib]: `Requesting visitor location` seems to have a double meaning. Would about just 'visitor location`?

Comment (willmore) [inline apiary.apib]: I see your point about `Requesting` being confusing.
Simply `Visitor location` is reasonable.

File: apiary.apib (hunk 4 above)
Comment (deiwin) [inline apiary.apib]: I disagree. This ties the sentence together by saying that the requesting visitor is the same current visitor who's data will be fetched.

Comment (willmore) [inline apiary.apib]: What other visitor would it be?

Comment (deiwin) [inline apiary.apib]: `Fetches current visitor's geographical location` doesn't say who has to do the requesting. And also makes the phrase `current visitor` ambiguous.

Comment (willmore) [inline apiary.apib]: `Current visitor` is already ambiguous (and should be reworded) per https://github.com/salemove/apiary_docs/pull/25/files/ff8154095ca81b7597e0e3a16af30e1b331eb1ad#r58648306

**Comment by ostap0207** [comment]
Updated

**Comment by ostap0207** [comment]
Any `+1s`?

**Comment by urmastalimaa** [comment]
+1

File: apiary.apib [hunk 6]
```diff
@@ -1306,6 +1306,36 @@ The ```note_update_method``` parameter takes either value ```replace``` which re
...
+    + Attributes
+
+        + country (required, string) - country name or "Unknown"
+        + country_code (required, string) - a two-letter (ISO 3166-1 alpha-2) code or "XX" if unknown
+        + city (required, string) - city name or "Unknown"
+
+    + Body
+
+        {
+          "country": "United States",
```
Comment (deiwin) [inline apiary.apib]: what about `href`? Shouldn't we include it in the response as well?

Comment (ostap0207) [inline apiary.apib]: Yes, it actually is like that, but forgot to add here.

**Comment by deiwin** [comment]
+1

**Comment by deiwin** [comment]
!squash

**Comment by deiwin** [comment]
> Automatic squash failed. Please squash manually

**Comment by ostap0207** [comment]
Done
